src/ApiCore/GapicClientTrait.php

```diff
@@ -83,57 +100,87 @@ private function setClientOptions(array $options)
             file_get_contents($options['clientConfigPath']),
             true
         );
-        $this->defaultCallSettings = CallSettings::load(
+        $this->retrySettings = RetrySettings::load(
             $options['serviceName'],
             $clientConfig,
             $this->pluck('retryingOverride', $options, false)
         );
+        $this->agentHeaderDescriptor = new AgentHeaderDescriptor([
+            'libName' => $this->pluck('libName', $options, false)
+            'libVersion' => $this->pluck('libVersion', $options, false)
+            'gapicVersion' => $options['gapicVersion']
+        ]);
         $descriptors = require($options['descriptorsConfigPath']);
         $this->descriptors = $descriptors['interfaces'][$options['serviceName']];
-        $this->transport = $transport instanceof ApiTransportInterface
+        $this->transport = $transport instanceof TransportInterface
             ? $transport
             : TransportFactory::build($options);
     }
 
-    private function configureCallSettings($method, array $optionalArgs)
+    /**
+     * @param string $method
+     *
+     * @return callable
+     */
+    private function getUnaryCallable($method)
```

Comment:
Should this perhaps have a more general signature, like accepting an options array? That would support future additions.

Reply:
I'm not crazy about having an undefined $options array. It would be more confusing, since it has the same name as the callable $options argument. We could just pass `Call` in, since $method is a part of that, but I felt like that was confusing as well. At least this way what the method needs is explicitly defined. Also, it's private so we can change this signature at any time.

Another option would be to remove `$method` and send the entire retrySettings config array to RetryMiddleware. But this seems a little hackier to me than what we have here.